api/system-dhcp/read

```diff
@@ -36,54 +39,82 @@ $ranges = array();
     }
-}
-$ret['status'] = $leases;
-
+    $ret['status'] = $leases;
 
-# Merge list of available interfaces with dhcp configuration
-$dhcp_db = new EsmithDatabase('dhcp');
-$ndb = new EsmithDatabase('networks');
-foreach (getNetworkInterfaces() as $i => $props) {
-    $range = $dhcp_db->getKey($i);
-    $nslabel = $ndb->getProp($i, 'nslabel');
-    $record = array("name" => $i ,"type" => "range", "nslabel" => $nslabel);
-    if ($range) {
-        foreach (array('DhcpDNS', 'DhcpWINS', 'DhcpNTP', 'DhcpTFTP') as $field) {
-            if (isset($range[$field])) {
-                if ($range[$field]) {
-                    $range[$field] = explode(",", $range[$field]);
-                } else {
-                    $range[$field] = array();
+    # Merge list of available interfaces with dhcp configuration
+    $dhcp_db = new EsmithDatabase('dhcp');
+    $ndb = new EsmithDatabase('networks');
+    foreach (getNetworkInterfaces() as $i => $props) {
+        $range = $dhcp_db->getKey($i);
+        $nslabel = $ndb->getProp($i, 'nslabel');
+        $record = array("name" => $i ,"type" => "range", "nslabel" => $nslabel);
+        if ($range) {
+            foreach (array('DhcpDNS', 'DhcpWINS', 'DhcpNTP', 'DhcpTFTP') as $field) {
+                if (isset($range[$field])) {
+                    if ($range[$field]) {
+                        $range[$field] = explode(",", $range[$field]);
+                    } else {
+                        $range[$field] = array();
+                    }
                 }
             }
+            $record['props'] = $range;
+        } else {
+            $record['props'] = getDefaults($props);
         }
-        $record['props'] = $range;
-    } else {
-        $record['props'] = getDefaults($props);
+        $ranges[] = $record;
     }
-    $ranges[] = $record;
-}
-$ret['configuration']['ranges'] = $ranges;
+    $ret['configuration']['ranges'] = $ranges;
 
-$hosts_db = new  EsmithDatabase('hosts');
-foreach ($hosts_db->getAll("local") as $key => $props) {
-    $record = array("name" => $key ,"type" => "local", "props" => $props);
-    $reservations[] = $record;
+    $hosts_db = new  EsmithDatabase('hosts');
+    foreach ($hosts_db->getAll("local") as $key => $props) {
+        $record = array("name" => $key ,"type" => "local", "props" => $props);
+        $reservations[] = $record;
+    }
+    $ret['configuration']['reservations'] = $reservations;
+
+    # JSON output
+    print json_encode($ret); 
 }
-$ret['configuration']['reservations'] = $reservations;
 
-# JSON output
-print json_encode($ret); 
+elseif ($action === 'scan') {
+    # scan the nic, check if the mac address is not already reserved
+    $nic = $data['nic'];
+    $hosts_db = new  EsmithDatabase('hosts');
+    $macs = array();
+
+    foreach ($hosts_db->getAll("local") as $key => $props) {
+        $mac = $hosts_db->getProp($key, 'MacAddress');
+        $macs[] = array_push ($macs,$mac);
+    }
+
+    exec("arp-scan -I $nic -lx", $output, $ret);
```

Comment:
It's better to protect user input with [escapeshellarg](https://www.php.net/manual/en/function.escapeshellarg.php).